sdk/storage/src/queue/mod.rs

```diff
@@ -78,6 +79,18 @@ pub trait MessageBodyRequired {
     fn message_body<'b>(&self) -> &str;
 }
 
+/// Sets both the message id and the pop receipt for deleting a message as per Azure specification.
+/// See
+/// [https://docs.microsoft.com/en-us/rest/api/storageservices/delete-message2](https://docs.microsoft.com/en-us/rest/api/storageservices/delete-message2)
+pub trait MessageSupport<'b> {
+    type O;
+    fn with_message(self, message: &'b Message) -> Self::O;
+}
+
+pub trait MessageRequired {
+    fn message<'b>(&self) -> &Message;
+}
```

Comment:
I like this idea. 
I would prefer if we use a *smaller* struct/trait for this (`Message` has a lot of fields that are useless in this context). 

Something like:

```rust
pub trait PopReceipt {
  fn message_id(&self) -> &str;  
  fn pop_receipt(&self) -> &str;
}
```

And we could implement `PopReceipt` for `Message`.

The API would become:

```rust 
pub trait PopRecepitSupport<'b> {
    type O;
    fn with_pop_receipt(self, message: &'b dyn PopReceipt) -> Self::O;
}

pub trait PopRecepitRequired {
    fn pop_receipt(&self) -> &dyn PopReceipt;
}
```

This way it could be used with `Message` as before but also, potentially, by a struct that has lesser fields. 

What do you think?

Reply:
That seems sensible, I can update this PR with that.

Reply:
I've updated the PR with that change. Let me know what you think!
Also, while testing this I found a bug, where pop_receipt can contain a '+' character. This gets interpreted as a space in the URL, meaning that the pop receipt became invalid. I have fixed this by adding percent encoding for that field.

Reply:
The part of this PR I think that I'm least sure about right now is using `Box<dyn PopReceipt>`, which I chose simply because I managed to convince the compiler that, that was correct. Still pretty new to rust so if that isn't appropriate then please let me know!

Reply:
It's ok! Whenever to own or to borrow is a subject of endless debate 😉, we'll address it if the time comes.